lib/auth/machineid/machineidv1/workload_identity_service.go

```diff
@@ -351,3 +370,153 @@ func (wis *WorkloadIdentityService) SignX509SVIDs(ctx context.Context, req *pb.S
 
 	return res, nil
 }
+
+func (wis *WorkloadIdentityService) signJWTSVID(
+	ctx context.Context,
+	authCtx *authz.Context,
+	clusterName string,
+	key *jwt.Key,
+	req *pb.JWTSVIDRequest,
+) (res *pb.JWTSVIDResponse, err error) {
+	var jti string
+	var spiffeID spiffeid.ID
+	defer func() {
+		evt := &apievents.SPIFFESVIDIssued{
+			Metadata: apievents.Metadata{
+				Type: events.SPIFFESVIDIssuedEvent,
+				Code: events.SPIFFESVIDIssuedSuccessCode,
+			},
+			UserMetadata:       authz.ClientUserMetadata(ctx),
+			ConnectionMetadata: authz.ConnectionMetadata(ctx),
+			Hint:               req.Hint,
+			SVIDType:           "jwt",
+			Audiences:          req.Audiences,
+		}
+		if err != nil {
+			evt.Code = events.SPIFFESVIDIssuedFailureCode
+		}
+		if !spiffeID.IsZero() {
+			evt.SPIFFEID = spiffeID.String()
+		}
+		if jti != "" {
+			evt.JTI = jti
+		}
+		if emitErr := wis.emitter.EmitAuditEvent(ctx, evt); emitErr != nil {
+			wis.logger.WarnContext(
+				ctx,
+				"Failed to emit SPIFFE SVID issued event.",
+				"error", emitErr,
+			)
+		}
+	}()
+
+	switch {
+	case req.SpiffeIdPath == "":
+		return nil, trace.BadParameter("spiffe_id_path: must be non-empty")
+	case !strings.HasPrefix(req.SpiffeIdPath, "/"):
+		return nil, trace.BadParameter("spiffe_id_path: must start with '/'")
+	case len(req.Audiences) == 0:
+		return nil, trace.BadParameter("audiences: must be non-empty")
+	}
+
+	// Perform authz checks. They must be allowed to issue the SPIFFE ID. Since
+	// this is a JWT SVID, there are no SANs to check.
+	if err := authCtx.Checker.CheckSPIFFESVID(
+		req.SpiffeIdPath,
+		[]string{},
+		[]net.IP{},
+	); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// Create a JTI to uniquely identify this JWT for audit logging purposes
+	jti, err = utils.CryptoRandomHex(jtiLength)
```

Comment:
Is the jti recommended to be just a random hex string rather than something like a UUID?

Reply:
As far as i'm aware within SPIFFE, there's no real requirement for it to take any specific form. I'd be open to switching it to a UUID if there's some advantage I'm not seeing.

Reply:
It prompted me to check the JWT spec and this is what I found in the [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519#page-10), pasting for reference:

> The "jti" (JWT ID) claim provides a unique identifier for the JWT.
> The identifier value MUST be assigned in a manner that ensures that
> there is a negligible probability that the same value will be
> accidentally assigned to a different data object; if the application
> uses multiple issuers, collisions MUST be prevented among values
> produced by different issuers as well.  The "jti" claim can be used
> to prevent the JWT from being replayed